packages/react-components/react-field/src/components/Field/useFieldStyles.ts

```diff
@@ -19,16 +19,51 @@ const useRootStyles = makeStyles({
   base: {
     display: 'grid',
     gridAutoFlow: 'row',
+    gridTemplateColumns: 'auto 1fr',
+    gridTemplateAreas: `
+      "label label"
+      "control control"
+      "validationIcon validationMessage"
+      "hint hint"
+    `,
     justifyItems: 'start',
   },
 
   horizontal: {
-    gridTemplateRows: 'auto auto auto auto',
-    gridTemplateColumns: '1fr 2fr',
+    gridTemplateColumns: '33% auto 1fr',
+    gridTemplateAreas: `
+      "label control control"
+      "label validationIcon validationMessage"
+      "label hint hint"
+      "label . ."
+    `,
   },
 
-  secondColumn: {
-    gridColumnStart: '2',
+  label: {
+    gridColumnStart: 'label',
+    gridColumnEnd: 'label',
+    gridRowStart: 'label',
+    gridRowEnd: 'label',
+  },
+
+  control: {
+    gridColumnStart: 'control',
+    gridColumnEnd: 'control',
+  },
+
+  validationIcon: {
+    gridColumnStart: 'validationIcon',
+    gridColumnEnd: 'validationIcon',
+  },
+
+  validationMessage: {
+    gridColumnStart: 'validationMessage',
+    gridColumnEnd: 'validationMessage',
+  },
+
+  hint: {
+    gridColumnStart: 'hint',
+    gridColumnEnd: 'hint',
```

Comment:
I put all the grid layout styles under `useRootStyles` just to make it easier to skim and understand.

I ended up not specifying row-start/row-end values for the single-row items. It's not quite the same as `grid-area`, but I figured it's not worth the extra classes to do so